cohortextractor/backends/databricks.py

```diff
@@ -0,0 +1,81 @@
+from cohortextractor.backends.base import BaseBackend, Column, MappedTable, QueryTable
+from cohortextractor.query_engines.spark import SparkQueryEngine
+
+
+class DatabricksBackend(BaseBackend):
+    backend_id = "databricks"
+    query_engine_class = SparkQueryEngine
+    patient_join_column = "patient_id"
+
+    patients = QueryTable(
+        columns=dict(
+            date_of_birth=Column("date"),
+        ),
+        # We're not (currently) provided with a patient table so we have to
+        # synthesize one by grabbing all the unique patients with their dates
+        # of birth from the prescriptions table. This obviously means that we
+        # don't know about patients which have never been prescribed a med.
+        # This is OK for our intial purposes.
+        query="""
+        SELECT
+            Person_ID AS patient_id, MAX(PatientDoB) AS date_of_birth
+        FROM
+            PCAREMEDS_pcaremeds
+        GROUP BY
+            Person_ID
+        """,
+    )
+
+    prescriptions = MappedTable(
+        source="PCAREMEDS_pcaremeds",
+        columns=dict(
+            patient_id=Column("integer", source="Person_ID"),
+            prescribed_dmd_code=Column(
+                "code", source="PrescribeddmdCode", system="dmd"
+            ),
+            processing_date=Column("date", source="ProcessingPeriodDate"),
+        ),
+    )
+
+    hospital_admissions = QueryTable(
+        columns=dict(
+            admission_date=Column("date"),
+            primary_diagnosis=Column("code", system="icd10"),
+            admission_method=Column("integer"),
+            episode_is_finished=Column("boolean"),
+            spell_id=Column("integer"),
+        ),
+        # HES data comes in distinct tables for each financial year which we
+        # union together to pretend its a single table.
+        query="\nUNION ALL\n".join(
+            # We join the APC (Admitted Patient Care) table with the MPS
+            # (Master Person Service) table so we can get the associated
+            # "person ID" which functions as our unique patient identifier that
+            # can be matched with the prescriptions data. We also join with the
+            # APC-OTR table (not sure what this stands for) so we can get a
+            # "spell ID" which allows us to see when multiple admissions
+            # correspond to the same hospitalisation "spell".
```

Comment:
Documenting the meaning of the abbreviations is _really_ helpful. Is `EPIKEY` a contraction for "episode"? If so, what's the difference between "episode" and "spell"?

Reply:
I didn't know, but your question prompted me to ask in Slack where I got a very helpful reply from Helen which I'll incorporate into the comments.